src/utils/svg.js

```diff
@@ -8,45 +11,88 @@ const runSVGO = primitiveSvg => {
   return retVal
 }
 
-// (Naively) Add Group to SVG
-// For schema, see: https://github.com/fogleman/primitive/blob/master/primitive/model.go#L86
-const patchSVGGroup = svg => {
-  const gStartIndex =
-    svg.match(/<path.*?>/).index + svg.match(/<path.*?>/)[0].length
-  const gEndIndex = svg.match(/<\/svg>/).index
-  const svgG = `<g filter='url(#c)' fill-opacity='.5'>`
-  return `${svg.slice(0, gStartIndex)}${svgG}${svg.slice(
-    gStartIndex,
-    gEndIndex
-  )}</g></svg>`
+const loadSVG = svg => {
+  return cheerio.load(svg, {
+    normalizeWhitespace: true,
+    xmlMode: true
+  })
 }
 
-// Add viewbox and preserveAspectRatio attributes as well as a Gaussian Blur filter to the SVG
-// When missing, add group (element with blur applied) using patchSVGGroup()
-// We initially worked with a proper DOM parser to manipulate the SVG's XML, but it was very opinionated about SVG syntax and kept introducing unwanted tags. So we had to resort to RegEx replacements
-const replaceSVGAttrs = (svg, { width, height, blur }) => {
-  let filter = ''
-  let blurStdDev = blur || 12
-  let blurFilterId = 'b'
-  let newSVG = svg
-
-  if (blur !== 0) {
-    if (svg.match(/<svg.*?><path.*?><g/) === null) {
-      blurStdDev = 55
-      newSVG = patchSVGGroup(newSVG)
-      blurFilterId = 'c'
-    } else {
-      newSVG = newSVG.replace(/(<g)/, `<g filter="url(#${blurFilterId})"`)
+// Prepare SVG. For now, this will just ensure that the viewbox attribute is set
+const prepareSVG = (svg, { width, height }) => {
+  const $ = loadSVG(svg)
+
+  const $svg = $('svg')
+
+  // Ensure viewbox
+  if (!$svg.is('[viewBox]')) {
+    if (!(width && height)) {
+      throw new Error(
+        `SVG is missing viewBox attribute while Width and height were not passed:\n\n${svg}`
+      )
     }
-    filter = `<filter id="${blurFilterId}"><feGaussianBlur stdDeviation="${blurStdDev}" /></filter>`
+    $svg.attr('viewBox', `0 0 ${width} ${height}`)
+  }
+
+  const $bgRect = $svg
+    .children(PRIMITIVE_SVG_ELEMENTS)
+    .filter('rect:first-child[fill]')
+
+  // Check if filling background rectangle exists
+  // This must exist for proper blur and other transformations
+  if (!$bgRect.length) {
+    throw new Error(
+      `The SVG must have a rect as first shape element which represents the svg background color:\n\n${svg}`
+    )
+  }
+
+  // Remove x and y attributes since they default to 0
+  // @todo test in rare browsers
```

Comment:
This is because there is no need to have a background rectangle which contains the exact width of the svg. The engine can calculate this on its own. So a `0 0 100% 100%` sizing should be totally fine.